runtime/vm/resolvesupport.cpp

```diff
@@ -2295,19 +2295,23 @@ resolveInvokeDynamic(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA callSite
 		if (vmThread->currentException != NULL) {
 			/* Already a pending exception */
 			result = vmThread->currentException;
-		} else if (result == NULL) {
+		} else if (NULL == result) {
 			setCurrentExceptionUTF(vmThread, J9VMCONSTANTPOOL_JAVALANGNULLPOINTEREXCEPTION, NULL);
-		} else { /* (result != NULL) */
-			/* Only write the value in if it is not null */
-			J9MemoryManagerFunctions *gcFuncs = vmThread->javaVM->memoryManagerFunctions;
-			J9Class *j9class = J9_CLASS_FROM_CP(ramCP);
-
-			/* Ensure that result array elements are written before the array reference is stored */
+			result = vmThread->currentException;
+		} else {
+			/* If the result is an array, ensure that the result array elements are written
+			 * before the array reference is stored.
+			 */
 			VM_AtomicSupport::writeBarrier();
```

Comment:
I'd rather have the write barrier unconditionally as it ensures that the exception object is safely published as well.  Given that we'll rarely take the resolve path, an extra writebarrier here is fine.

Reply:
Updated to invoke write barrier unconditionally.